opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/FeignTemplateToDataToSignMapper.java

```diff
@@ -68,6 +71,36 @@ public ConfirmConsentDataToSign mapToConfirmConsent(Map<String, Collection<Strin
         return new ConfirmConsentDataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType));
     }
 
+    public PaymentInitiationDataToSign mapToPaymentInitiation(Map<String, Collection<String>> headers, Instant instant, String body) {
+        String operationType = extractRequiredValue(headers, HttpHeaders.X_OPERATION_TYPE);
+        String xRequestId = extractRequiredValue(headers, HttpHeaders.X_REQUEST_ID);
+        String bankId = extractNonRequiredValue(headers, HttpHeaders.BANK_ID);
+        String fintechUserId = extractRequiredValue(headers, HttpHeaders.FINTECH_USER_ID);
+        String redirectOkUrl = extractRequiredValue(headers, HttpHeaders.FINTECH_REDIRECT_URL_OK);
+        String redirectNokUrl = extractRequiredValue(headers, HttpHeaders.FINTECH_REDIRECT_URL_NOK);
+
+        return new PaymentInitiationDataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType),
```

Comment:
Maybe we could use builder pattern here (and btw. PaymentInitiationDataToSign has builder declared), instead of introducing intermediate variables? Constructor seem to be quite long

Reply:
Lombok builder was added